test/lib/ansible_test/_internal/commands/integration/cloud/galaxy.py

```diff
@@ -96,45 +62,46 @@ def setup(self) -> None:
         """Setup cloud resource before delegation and reg cleanup callback."""
         super().setup()
 
+        name = 'galaxy-pulp'
         galaxy_port = 80
-        pulp_port = 24817
 
         ports = [
             galaxy_port,
-            pulp_port,
         ]
 
-        # Create the container, don't run it, we need to inject configs before it starts
         descriptor = run_support_container(
             self.args,
             self.platform,
-            self.pulp,
-            'galaxy-pulp',
+            self.image,
+            name,
             ports,
-            start=False,
+            aliases=[
+                name,
+            ],
+            start=True,
+            options=[
+                '--env-file',
+                os.path.join(
+                    os.path.dirname(__file__),
+                    'galaxy.env'
+                ),
+            ],
         )
 
         if not descriptor:
             return
 
-        pulp_id = descriptor.container_id
-
         injected_files = {
-            '/etc/pulp/settings.py': SETTINGS.format(host=descriptor.name).encode(),
-            '/etc/cont-init.d/111-postgres': SET_ADMIN_PASSWORD,
-            '/etc/cont-init.d/000-ansible-test-overrides': OVERRIDES,
+            '/etc/galaxy-importer/galaxy-importer.cfg': GALAXY_IMPORTER,
         }
-
         for path, content in injected_files.items():
             with tempfile.NamedTemporaryFile() as temp_fd:
                 temp_fd.write(content)
                 temp_fd.flush()
-                docker_cp_to(self.args, pulp_id, temp_fd.name, path)
-
-        descriptor.start(self.args)
+                docker_exec(self.args, descriptor.container_id, ['mkdir', '-p', os.path.dirname(path)], True)
+                docker_cp_to(self.args, descriptor.container_id, temp_fd.name, path)
```

Comment:
Are these files not needed before the container is started, to avoid a race condition?

Reply:
No, they are not needed before container start. The previous configuration needed to exist before start up, but since this one file is only read when `galaxy-importer` is called, and not by galaxy/pulp directly, it can be injected afterward.